[MRG+1] minor link fix for SAG paper.

iamprakashom opened the pull request:
#### Reference Issue

Fixes #7512 

updates link to SAG paper in the LinearRegression user guide.

SAG paper cited in _Reference_ section of docstring of `LogisticRegression`. 

updates link to SAG paper in _References_ section of `SAG Solvers`.

iamprakashom committed:
Link to SAG paper updated

link to SAG paper in `References` section updated (#7512).

jnothman commented:
LGTM

jnothman commented:
LGTM

jnothman commented:
Except
```
./sklearn/linear_model/logistic.py:219:80: E501 line too long (80 > 79 characters)
                                               shape=(n_samples, n_samples)), X)
```

iamprakashom committed:
Link to SAG paper updated

dead link of SAG paper in Logistic Regression user guide updated.
resolves #7512

iamprakashom committed:
SAG paper cited in docstring

SAG paper cited in `Reference` section of docstring(#7512).

iamprakashom commented:
@jnothman, done. kindly review. :smile:
